### ISSUE TITLE
Possible issue with timeseries query response schema

### ISSUE THREAD

**Issue by david-crespo** [oxidecomputer/omicron#5420]
The web console's generated method for `timeseries_query` takes the endpoint to return [nothing](https://github.com/oxidecomputer/console/blob/a7a11762ea742f99a28edad7bc80a9563219f874/app/api/__generated__/Api.ts#L6787-L6797) (`void`). I thought it was just a problem with the TS client generator, but the docs site also [doesn't like it](https://docs-bzmq3xgvn-oxidecomputer.vercel.app/api/timeseries_query) (response section does not expand), so it could be that the schema is actually bad and not just new to them.

Right off the bat, I notice we're directly returning some struct from the `oxql` crate — most likely we will want to make a view struct with a better name than `Table` and possibly a better ser/de situation.

https://github.com/oxidecomputer/omicron/blob/4c54580e797b042f131c5727388fbf8bbfb1e993/nexus/src/external_api/http_entrypoints.rs#L5658-L5666

**Comment by bnaecker** [comment]
> possibly a better ser/de situation.

What do you have in mind here?

**Comment by david-crespo** [comment]
Nothing yet, I just meant that if it turns out the difficulty in the docs site and TS client generator comes down to something unusual about the schema, it could be fixed by doing something in that view struct.

**Comment by david-crespo** [comment]
Ah. The answer was right there in the snippet. This endpoint returns an array when it should return an object with an `items` key. The TS client generator doesn't handle this case well, though it could and should, because we try to avoid doing this anyway. I haven't looked at the docs site yet, but it's probably the same deal. While looking into this I found a few other array-returning endpoints — they're all pretty new. I wonder if blocking direct array responses would be appropriate for `openapi-lint`. Maybe it's too strong of a restriction. @ahl?

I will take a stab at fixing this when I add a view struct in there so we can get a better name than `Table`.

```
/v1/system/networking/bfd-status
{
  title: 'Array_of_BfdStatus',
  type: 'array',
  items: { '$ref': '#/components/schemas/BfdStatus' }
}

/v1/system/networking/bgp-announce
{
  title: 'Array_of_BgpAnnouncement',
  type: 'array',
  items: { '$ref': '#/components/schemas/BgpAnnouncement' }
}

/v1/system/networking/bgp-routes-ipv4
{
  title: 'Array_of_BgpImportedRouteIpv4',
  type: 'array',
  items: { '$ref': '#/components/schemas/BgpImportedRouteIpv4' }
}

/v1/system/networking/bgp-status
{
  title: 'Array_of_BgpPeerStatus',
  type: 'array',
  items: { '$ref': '#/components/schemas/BgpPeerStatus' }
}

/v1/timeseries/query
{
  title: 'Array_of_Table',
  type: 'array',
  items: { '$ref': '#/components/schemas/Table' }
}
```

**Comment by bnaecker** [comment]
This was resolved by #6374
